nilearn/masking.py

```diff
@@ -513,6 +513,137 @@ def compute_multi_background_mask(data_imgs, border_size=2, upper_cutoff=0.85,
     return mask
 
 
+def compute_gray_matter_mask(target_img, threshold=.5,
+                             connected=True, opening=2, memory=None,
+                             verbose=0):
+    """ Compute a mask corresponding to the gray matter part of the brain.
+    The gray matter part is calculated through the resampling of MNI152
+    template Gray Matter mask onto the image
```

Comment:
I saw the code and you are using load_mni152_brain_mask which is not a gray matter mask but a whole brain mask. May be you need to check this.

Reply:
Maybe the name `template` is misleading but it's actually the MNI152 mask that we use, which is extracted from the template (I guess)